lib/api.common.js

```diff
@@ -657,39 +643,76 @@ api.common.eventEmitter = () => {
 
 api.common.cache = (
   // Extend Map interface total allocated size: map.allocated
+  options
 ) => {
   const cache = new Map();
+  const calcSize = (options && options.calcSize) || false;
+
+  if (calcSize) api.common.cache.withSize(cache);
+  else api.common.cache.withClear(cache);
+
+  return cache;
+};
+
+
+api.common.cache.withClear = (
+  cache // mixin to this Map
+) => {
+
+  cache.add = cache.set;
+
+  cache.del = cache.delete;
+
+  cache.clr = (
+    prefix, // string to compare with key start
+    fn //
+  ) => {
+    let key;
+    for (key of cache.keys()) {
+      if (key.startsWith(prefix)) {
+        cache.delete(key);
+        if (fn) fn(key);
+      }
+    }
+  };
+
+};
+
+api.common.cache.withSize = (
+  cache // mixin to this Map
+) => {
+
   cache.allocated = 0;
 
+  const dataSize = data => (data && data.lenght ? data.length : 0);
+
   cache.add = (key, val) => {
     if (cache.has(key)) {
       const prev = cache.get(key);
-      cache.allocated -= api.common.cacheSize(prev);
+      cache.allocated -= dataSize(prev);
     }
-    cache.allocated += api.common.cacheSize(val);
+    cache.allocated += dataSize(val);
     cache.set(key, val);
   };
 
   cache.del = (key) => {
     if (cache.has(key)) {
       const val = cache.get(key);
-      cache.allocated -= api.common.cacheSize(val);
+      cache.allocated -= dataSize(val);
     }
   };
 
-  cache.clearStartingWith = (
-    startsWith // string to compare with key start
+  cache.clr = (
+    prefix, // string to compare with key start
+    fn // function(key, val) to be called on each key (optional)
   ) => {
-    let key, val;
-    for ([key, val] of cache) {
-      if (key.startsWith(startsWith)) {
-        cache.allocated -= api.common.cacheSize(val);
+    cache.forEach((val, key) => {
```

Comment:
I thought the only difference was `for-of` vs `forEach`, but it actually needs `val`, so yeah, everything's okay here.